src/test/lib/optimizer/strategy/predicate_placement_rule_test.cpp

```diff
@@ -987,8 +1021,22 @@ TEST_F(PredicatePlacementRuleTest, DoNotMoveMultiPredicateSemiAndAntiJoins) {
 
     _apply_rule(_rule, _lqp);
 
+    EXPECT_TRUE(_optimization_context.is_cacheable());
     EXPECT_LQP_EQ(_lqp, expected_lqp);
   }
 }
 
+TEST_F(PredicatePlacementRuleTest, CheckCacheability) {
+  // clang-format off
+  _lqp = std::dynamic_pointer_cast<AbstractLQPNode>(PredicateNode::make(
+      or_(and_(equals_(_d_b, 1), equals_(_e_a, 10)), and_(equals_(_d_b, 2), equals_(_e_a, 1))),
+      JoinNode::make(JoinMode::Left, equals_(_d_a, _e_a),
+        _stored_table_d,
+        _stored_table_e)));
```

Comment:
(format + superfluous cast just as before)

Also, is there a need for such a complex predicate just in a single predicate node that could be pushed down?